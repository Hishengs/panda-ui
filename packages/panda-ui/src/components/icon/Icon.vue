<template>
  <span v-if="name" class="panda-icon" :style="wrapperStyle" @click="onClick">
    <i :data-feather="name" :style="style">
      <!--  -->
    </i>
  </span>
</template>

<script lang="ts" setup>
import { computed, onMounted } from 'vue';

defineOptions({
  name: 'panda-icon',
});

const props = defineProps({
  name: {
    type: String,
    default: '',
  },
  size: {
    type: [String, Number],
  },
  color: {
    type: String,
  },
});

const wrapperStyle = computed(() => {
  return {
    height: props.size ? parseInt(props.size + '', 10) + 'px' : '14px',
  };
});
const style = computed(() => {
  const style: {
    width: string;
    height: string;
    color?: string;
  } = {
    width: '14px',
    height: '14px',
  };
  if (props.color) {
    style.color = props.color.trim();
  }
  if (props.size) {
    const size = parseInt(props.size + '', 10);
    style.width = `${size}px`;
    style.height = `${size}px`;
  }
  return style;
});

onMounted(() => {
  if (window.feather) window.feather.replace();
});

const emit = defineEmits<{
  click: [e: MouseEvent];
}>();
const onClick = (e: MouseEvent) => {
  emit('click', e);
};
</script>
