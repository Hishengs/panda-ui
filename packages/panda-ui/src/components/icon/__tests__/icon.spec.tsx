import Icon from '../index';
import { describe, it, expect } from 'vitest';
import { shallowMount } from '@vue/test-utils';

describe('Icon', () => {
  it('should be default view', () => {
    const wrapper = shallowMount(Icon, {
      props: {
        name: 'user',
      },
    });
    expect(wrapper.html()).toMatchSnapshot();
  });

  it('proerty type', () => {
    const wrapper = shallowMount(Icon, {
      props: {
        name: 'user',
      },
    });
    expect(wrapper.html()).toMatchSnapshot();
    expect(wrapper.find('i').attributes('data-feather')).toBe('user');
  });

  it('proerty size', () => {
    const wrapper = shallowMount(Icon, {
      props: {
        name: 'user',
        size: '20',
      },
    });
    expect(wrapper.html()).toMatchSnapshot();
    expect(
      wrapper.find('i').attributes('style')?.includes('width: 20px;'),
    ).toBeTruthy();
  });

  it('proerty color', () => {
    const wrapper = shallowMount(Icon, {
      props: {
        name: 'user',
        color: '#000',
      },
    });
    expect(wrapper.html()).toMatchSnapshot();
    expect(
      wrapper.find('i').attributes('style')?.includes('color: rgb(0, 0, 0);'),
    ).toBeTruthy();
  });
});
