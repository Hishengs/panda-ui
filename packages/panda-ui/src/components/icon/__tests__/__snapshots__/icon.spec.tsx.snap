// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Icon > proerty color 1`] = `"<span class=\\"panda-icon\\" style=\\"height: 14px;\\"><i data-feather=\\"user\\" style=\\"width: 14px; height: 14px; color: rgb(0, 0, 0);\\"><!--  --></i></span>"`;

exports[`Icon > proerty size 1`] = `"<span class=\\"panda-icon\\" style=\\"height: 20px;\\"><i data-feather=\\"user\\" style=\\"width: 20px; height: 20px;\\"><!--  --></i></span>"`;

exports[`Icon > proerty type 1`] = `"<span class=\\"panda-icon\\" style=\\"height: 14px;\\"><i data-feather=\\"user\\" style=\\"width: 14px; height: 14px;\\"><!--  --></i></span>"`;

exports[`Icon > should be default view 1`] = `"<span class=\\"panda-icon\\" style=\\"height: 14px;\\"><i data-feather=\\"user\\" style=\\"width: 14px; height: 14px;\\"><!--  --></i></span>"`;
